### PR TITLE
Feature/decorate class

### PR DESCRIPTION
**Pull request by MosDevx**
### Changes Implemented:
  - Create a class Nameable  which is an Interface
  -  Refactored  Person class to inherit from  Nameable
  - Create a  base Decorator , NameableBaseDecorator
  - Create CapitalizeDecorator and TrimmerDecorator

Thank you

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -20,3 +26,6 @@ def is_of_age?
   end
   # rubocop:enable Naming/PredicateName
 end
+
+pers = Person.new(age: 10)
```
Comment (Davidosky007) [inline person.rb]: * Kindly ensure that your decorations work by trying out the code snippet in the project requirement, as i got an error when I try to test your decorations 
image
image

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @MosDevx 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
* No linter errors 👍 
* Good looking readme file 👍 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidsoky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 1 above)
Comment (Nemwel-Boniface) [inline person.rb]: - [ ] Just as mentioned by the previous reviewer, I am getting the very same error as him. Kindly look at why this is the case.

image

**Review by Nemwel-Boniface — CHANGES_REQUESTED** [review]
## Hello @MosDevx     :wave: , 
I'm Nemwel your reviewer for this PR :man_beard: 

Great job addressing some of the changes that were requested by a previous reviewer :clap: 
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights

Great job setting up the correct linters ✔️ 
Professional documentation ✔️ 
Correct workflow ✔️ 
Meaningful commit messages ✔️  


## ♻️ Required Changes ♻️

_Check the comments under the review._

## [Optional] suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding! :partying_face: 👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me **@Nemwel-Boniface** in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 1 above)
Comment (MosDevx) [reply person.rb]: Hey  @Nemwel-Boniface . 
I provided a class main.rb that has the code to test. It has a few changes cause of linter errors but it functions identically.

**Review by MeqdamAlqudah — APPROVED** [review]
Hi @MosDevx , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
image

## highlights 
:+1: no linter errors 
:+1: your code is working fine

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**